sled-agent/src/sim/server.rs

```diff
@@ -177,221 +171,43 @@ impl Server {
-            ),
-            SocketAddr::V6(a) => a,
-        };
-        let mut services = vec![NexusTypes::ServicePutRequest {
-            address: http_bound.to_string(),
-            kind: NexusTypes::ServiceKind::InternalDns,
-            service_id: Uuid::new_v4(),
-            sled_id: config.id,
-            zone_id: Some(Uuid::new_v4()),
-        }];
-
-        let mut internal_services_ip_pool_ranges = vec![];
-        if let Some(nexus_external_addr) = rss_args.nexus_external_addr {
-            let ip = nexus_external_addr.ip();
-
-            services.push(NexusTypes::ServicePutRequest {
-                address: config.nexus_address.to_string(),
-                kind: NexusTypes::ServiceKind::Nexus {
-                    external_address: ip,
-                    nic: NexusTypes::ServiceNic {
-                        id: Uuid::new_v4(),
-                        name: "nexus".parse().unwrap(),
-                        ip: NEXUS_OPTE_IPV4_SUBNET
-                            .nth(NUM_INITIAL_RESERVED_IP_ADDRESSES as u32 + 1)
-                            .unwrap()
-                            .into(),
-                        mac: MacAddr::random_system(),
-                    },
-                },
-                service_id: Uuid::new_v4(),
-                sled_id: config.id,
-                zone_id: Some(Uuid::new_v4()),
-            });
-
-            internal_services_ip_pool_ranges.push(match ip {
-                IpAddr::V4(addr) => {
-                    IpRange::V4(Ipv4Range { first: addr, last: addr })
-                }
-                IpAddr::V6(addr) => {
-                    IpRange::V6(Ipv6Range { first: addr, last: addr })
-                }
-            });
-        }
-
-        if let Some(external_dns_internal_addr) =
-            rss_args.external_dns_internal_addr
-        {
-            let ip = *external_dns_internal_addr.ip();
-            services.push(NexusTypes::ServicePutRequest {
-                address: external_dns_internal_addr.to_string(),
-                kind: NexusTypes::ServiceKind::ExternalDns {
-                    external_address: ip.into(),
-                    nic: NexusTypes::ServiceNic {
-                        id: Uuid::new_v4(),
-                        name: "external-dns".parse().unwrap(),
-                        ip: DNS_OPTE_IPV4_SUBNET
-                            .nth(NUM_INITIAL_RESERVED_IP_ADDRESSES as u32 + 1)
-                            .unwrap()
-                            .into(),
-                        mac: MacAddr::random_system(),
-                    },
-                },
-                service_id: Uuid::new_v4(),
-                sled_id: config.id,
-                zone_id: Some(Uuid::new_v4()),
-            });
-
-            internal_services_ip_pool_ranges
-                .push(IpRange::V6(Ipv6Range { first: ip, last: ip }));
-        }
-
-        let recovery_silo = NexusTypes::RecoverySiloConfig {
```

Comment:
All this stuff - silo setup, internal DNS, certs, IP ranges, etc, **still exists**, it has just been moved into the `run_standalone_server` function.

This is still useful for running the standalone sled agent binary, but tests likely want finer-grained control over services.